### ISSUE TITLE
Minimize window and then restore window duplicates plugins in vst rack

### ISSUE THREAD

**Issue by jonjamcam** [elieserdejesus/JamTaba#116]
first Jamtaba running normally.

![minimize1](https://cloud.githubusercontent.com/assets/15310433/11091659/7a6b5e32-885b-11e5-8c3d-2dc7d017083e.jpg)

 Then we minimize the window...and restore again:

![minimize2](https://cloud.githubusercontent.com/assets/15310433/11091674/9a042bb6-885b-11e5-8d14-6d352a7c5e6f.jpg)

And again.....minimize and restore.....

![minimize3](https://cloud.githubusercontent.com/assets/15310433/11091678/a375b7f0-885b-11e5-9163-b7c496ddc578.jpg)

**Comment by Ezeer** [comment]
waow ... you are a great bug finder !
![ugly](https://cloud.githubusercontent.com/assets/1283808/11091839/ddac0708-887e-11e5-84fd-a365aed4c0ae.png)

This one is ugly ...
Maybe we could code something that increase our paypal account instead of plug vst instances , lol !
We gonna fix that , of course .
Thx jon ! ( but you could have found it in the Rc3 no ? mmm..w.. ah ah ah  !)

**Comment by jonjamcam** [comment]
I will be faster next time! I Promise!

**Comment by Ezeer** [comment]
Ok , i tested on our actual git source code , and the bug is not present for me  .
I have loaded on vst , then two , added on sub channel , nothing , nada !
Can you test without that Kontakt vst please ? use an other vst to see if the problem is caused
by its window  .
thx

**Comment by jonjamcam** [comment]
just tested...strange....

![minimize bug](https://cloud.githubusercontent.com/assets/15310433/11093067/b18c9132-8866-11e5-84b4-bd6a717c11e3.jpg)

**Comment by Ezeer** [comment]
Duplicates or not Jon ?

**Comment by jonjamcam** [comment]
and again....

![minimize bug2](https://cloud.githubusercontent.com/assets/15310433/11093102/f5fcf7d0-8866-11e5-8072-e171b869db8a.jpg)

I think my json file stored that kontakt/izotope configuration and that is being triggered when minimize

**Comment by Ezeer** [comment]
Strange yeah ....
And the first vst on trk1 was not affected .
Try to see with only one track please ?

**Comment by jonjamcam** [comment]
not exactly duplicate......I changes to Tracks and restart so json updates....now it's triggering the tracks

![minimize bug3](https://cloud.githubusercontent.com/assets/15310433/11093191/79de82b2-8867-11e5-8537-845653796589.jpg)

**Comment by jonjamcam** [comment]
Obviously it's reading the json file and activating the corresponding rack when it restores after minimize.

**Comment by Ezeer** [comment]
Okay , i have the same bug here .
You are right , the bug appeared when i closed then relaunched Jamtaba .
So it is an error of read or write in the Json file , i'll inspect that  .
Very nice bug my dear !

**Comment by jonjamcam** [comment]
only doing my job!

**Comment by elieserdejesus** [comment]
Owww my good!  This is a BUG, with upper case :)

Are you working in this Ezee?

**Comment by Ezeer** [comment]
not yet , but i can take it .
I take it ?

**Comment by elieserdejesus** [comment]
Please Ezee, take it!

Ezee, I see the mess here: Look at MainWindowStandalone.cpp, line 25:
```
void MainWindowStandalone::showEvent(QShowEvent *){
    //wait 50 ms before restore the plugins list to avoid freeze the GUI in hidden state while plugins are loading
    QTimer::singleShot(50, this, &MainWindowStandalone::restorePluginsList);
}

```

**Comment by Ezeer** [comment]
ok , thanks i check that .

**Comment by Ezeer** [comment]
Done .
Mooved the reload vst list in window constructor .
Can be improved by modifying restorePluginsList : 
restorePluginsList (bool fromSettings) ; // true read from memory , false from disk ...

**Comment by jonjamcam** [comment]
that was fast!

**Comment by Ezeer** [comment]
Yeah , elieser pointed to me the good place to look after .
So ... next bug is ..?
lol

**Comment by jonjamcam** [comment]
what do you thin guys about doing a little jam session once 2.0.5 is finished to celebrate?

I bring the girls! lol

**Comment by Ezeer** [comment]
good idea !
But i've got to setup my studio before ...
I'm in the saloon since 14 days ... hips .. sorry .
lol

**Comment by Ezeer** [comment]
@jonjamcam  the green girl is for me !

**Comment by jonjamcam** [comment]
let's let her choose ;)

**Comment by Ezeer** [comment]
Deal !
ah ah ah ah ...

**Comment by elieserdejesus** [comment]
Ha ha! Girls in ninjam? What?????

I can't jamming at moment because I have a lot of connection problems in my house and I don't know the cause yet (bad cables or bad modem/router).

I'm using internet in my job, but here in the job I don't have my instruments :)

**Comment by Ezeer** [comment]
" I'm using internet in my job, but here in the job I don't have my instruments :) "
No problem , you'll be hour singer .
Don't be afraid , i have a song your boss will adore : 
" I love my job , lord , i love my Job , because my job is the coolest one , my god ! "

wait .... hey !!! She's with me man ! Look the orange thing ! Man ...
![addict](https://cloud.githubusercontent.com/assets/1283808/11097296/ebf05c68-889e-11e5-8691-8b017b0e2fd9.jpg)

**Comment by jonjamcam** [comment]
well, without you Elieser it wouldn't be the same. I think Ezee we should put the party on hold until Elieser can join, don't you think?

maybe for 2.0.6?

**Comment by Ezeer** [comment]
Forget me .... the green girl made her choice already .
( but perhaps she will be fooled by my long hairs ? )
so .... may be !

**Comment by elieserdejesus** [comment]
Maybe you can help to fix my internet problems. 
When I look at my router config page I see a lot of Rx errors. Many network packets are dropped. I'm suspecting from my cables because we have a long rain period here, so the external cables are a little wet. Or maybe the problem is my router.

How I can test my cables? Any idea?

**Comment by jonjamcam** [comment]
Of course...girls on ninjam!!!!

-![Sharon](http://ninbot.com/files/imagecache/large/pictures/photos/IMG_0470.JPG)

http://ninbot.com/content/impro-septembre-2012

She's a "badbassgirl"

**Comment by elieserdejesus** [comment]
Just one :)  Ah, I see another girl singing, very good singer. But she rarely jamming and don't remember the name or the country.

**Comment by jonjamcam** [comment]
"How I can test my cables? Any idea"

first plug your computer directly yo the modem (I'm supposing you have a modem and a router). That way you can tell if the flaw is inside the house or outside.

**Comment by Ezeer** [comment]
>RX Errors are typically caused by one or more of the following:

>Positive values in RX-ERR counter mean that the NIC received malformed Ethernet frames from the transmitting switch port, and data integrity could not be validated during frame's cyclic redundancy check (CRC) . The root cause of this is usually a bad cable, or a bad interface on either the machine or the switch.
**NIC speed / duplex mis-match with the connecting port on the switch/router.**
High or critical performance rated IPS protections are set to Prevent or Detect.
High number of rules in Security and/or in NAT policy.

**Comment by elieserdejesus** [comment]
I do this test Ezee, don't work too. So the problem is router or external cable.

**Comment by jonjamcam** [comment]
Last time I had bad connection It was my router so I bought a new one...but the technician had to put an attenuator. He explained a signal to high in power can lead to malfunction too. There's a range of dB not to be exceeded.
